R/dplyr.R

```diff
@@ -157,8 +157,21 @@ slice.dm <- function(.data, ...) {
 }
 
 #' @export
-slice.zoomed_dm <- function(.data, ...) {
-  replace_zoomed_tbl(.data, slice(get_zoomed_tbl(.data), ...))
+slice.zoomed_dm <- function(.data, idx) {
+  if (is_missing(idx)) return(.data)
+
+  if (!inherits(idx, "integer")) {
+    # numeric vector only accepted if identical to itself as integer
+    if (!inherits(idx, "numeric") || !all(as.integer(idx) == idx)) abort_need_int(class(idx))
+  }
+  # FIXME: will be easier if we have an extra row in `def` for the zoomed table
+  if (cdm_has_pk(.data, !!orig_name_zoomed(.data))) {
```

Comment:
Can we init `tracked_keys <- NULL` here and return `replace_zoomed_tbl(..., tracked_keys)` unconditionally?